nautobot/extras/views.py

```diff
@@ -1687,6 +1687,15 @@ def post(self, request, pk):
                     return render(request, "extras/job_approval_confirmation.html", {"scheduled_job": scheduled_job})
                 scheduled_job.approved_by_user = request.user
                 scheduled_job.approved_at = timezone.now()
+                if (
+                    scheduled_job.interval == JobExecutionType.TYPE_IMMEDIATELY
+                    and scheduled_job.start_time < timezone.localtime()
+                ):
+                    # This is needed because otherwise multiple jobs will be scheduled, not just one.
+                    scheduled_job.one_off = True
```

Comment:
Should this have been set in `create_schedule()`? Right now in that function we set `one_off=(interval == JobExecutionType.TYPE_FUTURE)` and I wonder if that's incorrect.